source/dofs/dof_tools_constraints.cc

```diff
@@ -1584,12 +1660,152 @@ namespace DoFTools
 
                       case FiniteElementDomination::neither_element_dominates:
                       {
-                        // we don't presently know what exactly to do here.
-                        // it isn't quite clear what exactly we would have
-                        // to do here. sit tight until someone trips over
-                        // the following statement and see what exactly is
-                        // going on
-                        Assert (false, ExcNotImplemented());
+                        // make sure we don't get here twice from each cell
+                        if (processed_p_faces.find(cell->face(face)) != processed_p_faces.end())
+                          break;
+
+                        processed_p_faces.insert(cell->face(face));
+
+                        // our best bet is to find the common space among other
+                        // FEs in FECollection and then constrain both FEs
+                        // to that one.
+                        // More precisely, we follow the strategy outlined on
+                        // page 17 of the hp paper:
+                        // First we find the dominant FE space S.
+                        // Then we divide our dofs in master and slave such that
+                        // I^{face,master}_{S^{face}->S} is invertible.
+                        // And finally constrain slave dofs to master dofs based
+                        // on the interpolation matrix.
+
+                        const unsigned int this_fe_index = cell->active_fe_index();
+                        const unsigned int neighbor_fe_index = neighbor->active_fe_index();
+                        const unsigned int dominating_fe_index =
+                          internal::find_dominating_fe(dof_handler, {this_fe_index,neighbor_fe_index});
```

Comment:
good enough for the moment -- but initialization of a std::set only works this way (with curly braces) in C++11. You'll have to adjust this for the final version of the patch.